### PR TITLE
Update docs of backend_manager

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

File: lib/matplotlib/backend_managers.py [hunk 1]
```diff
@@ -43,11 +43,11 @@ class ToolManager:
 
     Attributes
     ----------
-    figure : `Figure`
-    keypresslock : `widgets.LockDraw`
-        `LockDraw` object to know if the `canvas` key_press_event is locked
-    messagelock : `widgets.LockDraw`
-        `LockDraw` object to know if the message is available to write
+    figure : `~matplotlib.figure.Figure`
```
Comment (anntzer) [inline lib/matplotlib/backend_managers.py]: Lets stick to just \`.Figure\`?  (other objects may be more obscure and thus fully qualified, but .Figure doesn't need the full module name...)

Comment (timhoffm) [reply lib/matplotlib/backend_managers.py]: That's not what our docstring guide says https://matplotlib.org/devdocs/devel/documenting_mpl.html#referencing-types. (But hasn't been consistent throughout the file anyway. So, reverted for now.)

Comment (anntzer) [reply lib/matplotlib/backend_managers.py]: I could rather change the guide in that case...

File: lib/matplotlib/backend_managers.py [hunk 2]
```diff
@@ -168,7 +172,8 @@ def get_tool_keymap(self, name):
 
         Returns
         -------
-        list : list of keys associated with the Tool
+        list
```
Comment (anntzer) [inline lib/matplotlib/backend_managers.py]: "list of str"?

Comment (timhoffm) [reply lib/matplotlib/backend_managers.py]: Done.

**Pull request merged**